app.rb

```diff
@@ -0,0 +1,50 @@
+require 'sinatra'
+require 'sinatra/reloader'
+require './lib/peep'
+require './lib/sign_up'
+require './database_connection_setup'
+
+class Chitter < Sinatra::Base
+
+  configure :development do 
+    register Sinatra::Reloader
+  end
+
+  get '/' do 
+    erb :index
+  end
+
+  get '/sign_up' do
+    erb :sign_up
+  end
+
+  post '/confirm_sign_up' do
+    SignUp.create(email: params[:email], password: params[:password], user_name: params[:user_name], handle: params[:handle])
+    @user_info = SignUp.all.reverse.first
+    erb :confirm_sign_up
+  end
+
+  get '/log_in' do
+    erb :log_in
+  end
+
+  post '/confirm_log_in' do
+    if SignUp.log_in_verified?(email: params[:email], password: params[:password])
+      @user_info = SignUp.all.reverse.first
```

Comment:
What you want to retrieve here is actually the user info of the *user who just signed in*, so you have more than one solution:
 * change `SignUp.log_in_verified?` so it returns the `SignUp` object corresponding to this account instead — the condition then changes to verify it is not `nil`
 * add a new method to `SignUp` to retrieve an account by email, so you can use it in the controller:
 ```ruby
@user_info = SignUp.get_by_email(params[:email])
```

If you do this, you might also want to save the user ID into the `session` object so you keep a trace of who's logged in across requests (by storing this session ID, you can then retrieve the user profile in further HTTP requests made to the app)